pandas/core/arrays/numpy_.py

```diff
@@ -162,8 +162,14 @@ def _from_sequence(cls, scalars, dtype=None, copy=False):
         result = np.asarray(scalars, dtype=dtype)
         if copy and result is scalars:
             result = result.copy()
+        # Primarily for StringArray
+        result = cls._from_sequence_finalize(result, copy=copy)
         return cls(result)
 
+    @staticmethod
+    def _from_sequence_finalize(values, copy):
+        return values
```

Comment:
Why is the different name needed, and can't it just be `_from_sequence`?

Reply:
This lets StringArray customize a bit of the logic in the *middle* of `_from_sequence`. Without this StringArray._from_sequence raised on the `super()._from_sequence`, since the init would raise. This lets us still reuse PandasArray._from_sequence.

Happy to have suggestions for a better name. I didn't put any thought into this one. `_finalize` isn't the best since it's not happening after `_from_sequence`, it's in the middle.

Reply:
So the call chain is roughly

```
StringArray._from_sequnce
  -> PandasArray._from_sequnce
  -> StringArray._from_sequnce_finalize
-> StringArray._from_sequence  (after the super finishes)
```

hopefully that makes sense.

Reply:
Ah, I didn't notice that it still called super in the middle, understand now. 

But, I would personally just not call super and implement all in `StringArray._from_sequence`. Yes, this gives a little bit of duplication (but really not a lot IMO), but I think that is better than the additional method name and harder to understand call chain.

Reply:
Renamed to `_coerce_from_sequence_values`

Reply:
OK, I've inlined things. Apologies for the force push, I'm not sure what happened.